### DISCUSSION TITLE
No Linux??

### DISCUSSION THREAD

**Discussion by djbradster** [gtxaspec/wz_mini_hacks#234] [Q&A]
Would love to do this on all my Wyze cams, but apparently Linux is needed to compile for the V2's.   I don't have a Linux box - any simple way to do this in Windows?

**Comment by baudneo** [comment]
Windows subsystem for linux

**Reply by Bob-AOL** [reply]
Is there a download source somewhere for a pre-compiled V2 SD-ROOT directory for Windows users to then be transferred to an SD card for subsequent editing / access on the camera itself via Nano or VI?

Having to install a WSL (or whatever) just to compile the script seems a bit ridiculous.

**Reply by Bob-AOL** [reply]
Nevermind. Piece o' cake. Thanks for the suggestion.

Microsoft Windows [Version 10.0.19043.1826]
(c) Microsoft Corporation. All rights reserved.

C:\WINDOWS\system32>wsl --install
Installing: Virtual Machine Platform
Virtual Machine Platform has been installed.
Installing: Windows Subsystem for Linux
Windows Subsystem for Linux has been installed.
Downloading: WSL Kernel
Installing: WSL Kernel
WSL Kernel has been installed.
Downloading: Ubuntu
The requested operation is successful. Changes will not be effective until the system is rebooted.

C:\WINDOWS\system32>

**Reply by Bob-AOL** [reply]
'Piece o' cake' may have been just slightly premature.

After what appeared to be a flawless Ubuntu sub-installation, I just spent the past three hours un-bricking what had previously been a very nicely tuned laptop. Not the resulting solution I was hoping for -- or intend to attempt again.

Back to the drawing board once I clean up the broken glass and take stock of how bad the HDD is scrambled.

All to hack a little POS hobby cam. LOL

If it weren't fun, eh?

**Reply by baudneo** [reply]
I don't use windows but I am surprised that installing WSL would "brick" your laptop. You could also do it from a bootable usb that runs a "live" version of linux.

**Reply by Bob-AOL** [reply]
The laptop was rendered inoperable and wouldn't even post. Several hard power cycles finally threw something into the mix that brought Windows up in a completely unfamiliar recovery mode which included a disclaimer about the possibility of a successful recovery. Three hours later it rendered a desktop while the HDD spun away at 100% for another hour resulting in the discovery of over 100 registry errors.

I just attempted it again on a more robust system with the same virtual machine error as encountered with the laptop and simply uninstalled it before 'pressing any key to continue' suggestion this time around. The system recovered. Another solution for either system looks to be necessary for whatever reason.

**Reply by djbradster** [reply]
> 



> Is there a download source somewhere for a pre-compiled V2 SD-ROOT directory for Windows users to then be transferred to an SD card for subsequent editing / access on the camera itself via Nano or VI?
> 
> Having to install a WSL (or whatever) just to compile the script seems a bit ridiculous.

I can give wls a shot, but I love this question.   Why can't we just download the pre-compiled version?

**Reply by Bob-AOL** [reply]
Sure wish it was as easy as it seems like it should be. My error is a virtual machine error on two separate machines -- both of which have virtualization enabled.

I could have 50 cameras up and running with a pre-compiled version in the time I've spent pissin' up a rope with this.

**Reply by baudneo** [reply]
> > 
> 
> 
> 
> > Is there a download source somewhere for a pre-compiled V2 SD-ROOT directory for Windows users to then be transferred to an SD card for subsequent editing / access on the camera itself via Nano or VI?
> > 
> > Having to install a WSL (or whatever) just to compile the script seems a bit ridiculous.
> 
> I can give wls a shot, but I love this question.   Why can't we just download the pre-compiled version?

The v2 firmware from wyze is downloaded and then a firmware or kernel is compiled from the wyze package. It may be a legal or licensing issue as to why it is not precompiled, sometimes things aren't as cut and dried as you wish they would be. 

I myself prefer compiling the image locally instead of risking Wyze issuing a DMCA or takedown request and we loose the repo all together.

**Reply by baudneo** [reply]
> Sure wish it was as easy as it seems like it should be. My error is a virtual machine error on two separate machines -- both of which have virtualization enabled.
> 
> I could have 50 cameras up and running with a pre-compiled version in the time I've spent pissin' up a rope with this.

Get an 8gb flash drive, burn Ubuntu 22.04 onto it using Rufus. Boot from the usb and select "live" mode. Compile the image and you're done. Your host machine will be untouched as the live Ubuntu is ran off of the flash drive and RAM.

I'll write up a how to for the live version in the next few days when I have some time. 

It's extremely easy to get the image compiled tbh.

**Reply by Bob-AOL** [reply]
"Live" Ubuntu is next. And where did I miss wz mini being compiled from a Wyze package? 

As for DMCA non-compliance being an issue for someone compiling the hack and making it available, that seems unlikely considering the Wyze kernel itself is open source and the existing Wyze firmware remains unmodified though out the process. I highly doubt the copywrite police are going to be banging down any doors or issuing cease and desist threats to the small community of folks playing around with this stuff...., but I may be wrong.

Anyway, the assistance is greatly appreciated. Thanks.

**Reply by gtxaspec** [reply]
@baudneo is correct, our policy is not to host any firmware from the vendor.  The V2 being an older model has a different installation process compared to the T31 devices. This means we need to unpack the firmware, and then repack it with modifications.  Working on potential solutions for this now.

**Reply by Bob-AOL** [reply]
Ack. Perhaps some additional clarification in the readme would be helpful in distinguishing between the relative ease of the V3 implementation simply booting from an SD card and what can obviously be a rather frustrating experience for a non-Linux person attempting the required modifications to earlier firmware. Apples and oranges, really. So are the consequences of misinterpretations of the differences between the two implementations. 

Don't get me wrong. I'm tremendously appreciative of the time and effort that has gone into the project. That said, re-packaging the firmware into the camera with Linux is just slightly different than simply copying a few files to an SD card with Windows and re-booting.

A bit more detail and clarification in the readme between the two may go a long way toward minimizing unnecessary Q&A traffic along with generating fewer frustrated users trying to make either approach work for their particular situation.

**Reply by baudneo** [reply]
Fork it, compile it and distribute it yourself than if you are sure about the things you are saying.

It took wyze along time to release their firmware and licenses even after deploying their products. Also, recently I saw posts on Reddit about wyze making newer firmwares hostile towards rtsp. 

Does that sound like a company you want to piss off?

**Reply by Bob-AOL** [reply]
Spare me, dude. The only thing I'm sure about is it only works on one of three of my V2s and won't boot at all on the other two. Sheesh.

**Comment by djbradster** [comment]
Wow...pretty sure it's just easier to throw these wyze cams in the trash
and buy the Tapo cams that come ONVIF compliant.  They're actually cheaper,
and 2K.  Would be nice if the complete simplified instructions were just
written out instead of "compile on linux"...but why do that???

**Reply by Bob-AOL** [reply]
It's a hobby cam. The challenge is obviously part of the fun. Getting it to work as a more useful and capable device is gratifying as a hobbyist. (I'd say most of us also have higher quality 'real' cameras in our systems as well. I certainly do.)

Successful efforts provide wired ethernet and POE for these cams. The V3s take all of 10 minutes to do and I'll take any you want to throw away. Same for your V2s as I'm sure they're just as easy -- "once you know how". 

Sometimes the steepest learning curves apply to performing the simplest tasks. That said, I _am_ going on day three of trying to apply the hack to a V2. There should obviously be an easier method than what is currently available.

**Reply by baudneo** [reply]
> Wow...pretty sure it's just easier to throw these wyze cams in the trash
> and buy the Tapo cams that come ONVIF compliant.  They're actually cheaper,
> and 2K.  Would be nice if the complete simplified instructions were just
> written out instead of "compile on linux"...but why do that???
> 
> On Fri, Aug 12, 2022 at 8:47 PM baudneo ***@***.***> wrote:
> 
> > Sure wish it was as easy as it seems like it should be. My error is a
> > virtual machine error on two separate machines -- both of which have
> > virtualization enabled.
> >
> > I could have 50 cameras up and running with a pre-compiled version in the
> > time I've spent pissin' up a rope with this.
> >
> > Get an 8gb flash drive, burn Ubuntu 22.04 onto it using Rufus. Boot from
> > the usb and select "live" mode. Compile the image and you're done. Your
> > host machine will be untouched as the live Ubuntu is ran off of the flash
> > drive and RAM.
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/gtxaspec/wz_mini_hacks/discussions/234#discussioncomment-3388166>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/A2OUT2IIWDWUU7SBLPOJP23VY4LGRANCNFSM56DVXBRQ>
> > .
> > You are receiving this because you authored the thread.Message ID:
> > ***@***.***>
> >
> 

Lol! The instructions don't just say "compile on Linux" it says to run a custom made shell script that has been crafted and written for you so you don't mess things up.

The gall and audacity of you to say that and whine about things says quite alot about your character.

Be grateful someone tookt heir time to put this together and make it as easy as it is. Or better yet, take the original repos and make something like this yourself.

**Reply by Bob-AOL** [reply]
Well, I'm now going on three days of trying to apply these enhanced features to the V2 cameras and learning quite a bit in the process. I hope your comments weren't directed toward my occasional expressions of frustration as I slowly move forward. I'm quite appreciative of any assistance I can scrounge up around here!

I did manage to get a live Ubuntu Desktop going today -- on what ended up being a USB 1.0 thumb drive. LOL Yeah, it really screams. Didn't get to figuring out how to compile the necessary stuff yet and could use a tip or two there. Worked with the graciously supplied zip file a bit to kind of get my feet wet with the approach to the V2. The wifi connectivity is problematic on a couple levels, not the least of which is when it's toggled off in the config, it doesn't seem to want to come back without a fair amount of help..., certainly not  automatically just by disabling the ethernet in the configuration -- if you can get to the configuration again without pulling the SD card. Jumped through a lot of hoops today trying to work around that with no luck. An adapter also seems to be required to go from the mini USB male on the cable to a male USB 2.0 male for the corresponding port on the cam to establish both power and ethernet functionality from the same connection..

**Comment by gtxaspec** [comment]
Being that the camera natively runs Linux, the tools to create the firmware package also need to be run on Linux.  
If someone is familiar enough with windows, perhaps the script can be ported to windows.

**Reply by Bob-AOL** [reply]
See comments above.

**Comment by gtxaspec** [comment]
new installation method in testing:
see https://github.com/gtxaspec/wz_mini_hacks/issues/241

**Comment by Bob-AOL** [comment]
After a ton of reading and lots of work (trial and error), I'd hafta say the WSL method is probably the most efficient solution in the long run for a Windows box **_IF_** your system conforms to the necessary prerequisites. No fun at all if you learn the hard way that it doesn't.

After nearly two weeks of peckin' at it, I finally have Ubuntu co-existing with Windows as simultaneously operational systems. Very cool, but certainly not for everyone.

The less risky (and potentially less painful) way is to create a "live" Ubuntu flash drive as also previously suggested. It can be a slow way to go about everything, but you aren't risking unforeseen consequences to your existing Windows environment. Moving files back and forth isn't as straightforward as it appears it should be, and can potentially be problematic when manipulating files between the two environments.

The only reason I went the distance on this was to see of I can get my two problem V2s running wz_mini by sticking strictly to the Linux environment for the entire process -- the thought being that maybe the flash memory on the two cams is somehow more susceptible to any anomalies introduced by Windows decompression and copying than the flash in the cam that loads wz_mini without issue.

Probably a long shot, but that's about all I haven't tried.

**Reply by Bob-AOL** [reply]
FWIW, doing the whole process (original method) in a Linux environment made no difference for my two problem cameras.
